src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp

```diff
@@ -88,33 +121,39 @@ Primer3Dialog::Primer3Dialog(const Primer3TaskSettings& defaultSettings, ADVSequ
             repeatLibraries[i].second = QFileInfo(QString(PATH_PREFIX_DATA) + ":" + repeatLibraries[i].second).absoluteFilePath().toLatin1();
     }
 
-    {
-        QPair<QString, QByteArray> library;
-        foreach (library, repeatLibraries) {
-            ui.combobox_PRIMER_MISPRIMING_LIBRARY->addItem(library.first);
-            ui.combobox_PRIMER_INTERNAL_OLIGO_MISHYB_LIBRARY->addItem(library.first);
-        }
+    for (const auto& library : repeatLibraries) {
+        combobox_PRIMER_MISPRIMING_LIBRARY->addItem(library.first);
+        combobox_PRIMER_INTERNAL_MISHYB_LIBRARY->addItem(library.first);
     }
 
     int spanIntronExonIdx = -1;
-
-    for (int i = 0; i < ui.tabWidget->count(); ++i) {
-        if (ui.tabWidget->tabText(i).contains("Intron")) {
+    for (int i = 0; i < tabWidget->count(); ++i) {
+        if (tabWidget->tabText(i).contains("Intron")) {
             spanIntronExonIdx = i;
             break;
         }
     }
 
     if (spanIntronExonIdx != -1) {
         SplicedAlignmentTaskRegistry* sr = AppContext::getSplicedAlignmentTaskRegistry();
-        ui.tabWidget->setTabEnabled(spanIntronExonIdx, sr->getAlgNameList().size() > 0);
+        tabWidget->setTabEnabled(spanIntronExonIdx, sr->getAlgNameList().size() > 0);
     }
 
     reset();
 }
 
-Primer3TaskSettings Primer3Dialog::getSettings() const {
-    return settings;
+Primer3Dialog::~Primer3Dialog() {
+    if (settings != nullptr) {
```

Comment:
no need for null check - you can delete nulls safely

Reply:
OK